apps/src/lib/kits/maker/ui/SetupGuide.jsx

```diff
@@ -1,417 +1,109 @@
-              });
-          }}
-        />
-      );
-    }
-
-    if (isCodeOrgBrowser() || shouldUseWebSerial()) {
-      return <SetupChecklist webSerialPort={webSerialPort} />;
-    }
-    return (
-      <Provider store={store}>
-        <Downloads />
-      </Provider>
-    );
-  }
-}
-
-class Downloads extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {platform: Downloads.platformFromHash()};
-  }
-
-  componentDidMount() {
-    window.addEventListener('hashchange', this.onHashChange);
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('hashchange', this.onHashChange);
-  }
-
-  static platformFromHash() {
-    const hash = window.location.hash.slice(1);
-    if ([WINDOWS, MAC, LINUX, CHROMEBOOK].includes(hash)) {
-      return hash;
-    } else if (isWindows()) {
-      return WINDOWS;
-    } else if (isOSX()) {
-      return MAC;
-    } else if (isLinux()) {
-      return LINUX;
-    }
-    return WINDOWS;
-  }
-
-  onHashChange = () => {
-    this.setState({platform: Downloads.platformFromHash()});
-  };
-
-  onPlatformChange = platform => {
-    window.location.hash = '#' + platform;
-    this.setState({platform});
-  };
-
-  render() {
-    const {platform} = this.state;
-    return (
-      <div>
-        <ToggleGroup selected={platform} onChange={this.onPlatformChange}>
-          <button type="button" value={WINDOWS}>
-            <FontAwesome icon="windows" /> {i18n.windows()}
-          </button>
-          <button type="button" value={MAC}>
-            <FontAwesome icon="apple" /> {i18n.mac()}
-          </button>
-          <button type="button" value={LINUX}>
-            <FontAwesome icon="linux" /> {i18n.linux()}
-          </button>
-          <button type="button" value={CHROMEBOOK}>
-            <FontAwesome icon="chrome" /> {i18n.chromebook()}
-          </button>
-        </ToggleGroup>
-        {WINDOWS === platform && <WindowsDownloads />}
-        {MAC === platform && <MacDownloads />}
-        {LINUX === platform && <LinuxDownloads />}
-        {CHROMEBOOK === platform && <ChromebookInstructions />}
-        <h2>{i18n.support()}</h2>
-        <SafeMarkdown markdown={i18n.debugMakerToolkit()} />
-        <SafeMarkdown markdown={i18n.contactGeneralSupport()} />
-      </div>
-    );
+const setupGuideContent = {
+  microbit: {
+    id: 'microbit-description',
+    title: applabI18n.makerSetupMicrobitTitle(),
+    href: 'https://microbit.org/',
+    imgSrc: '../assets/maker/microbit-drawing-green.png',
+    description: applabI18n.makerSetupMicrobitDescription(),
+    imgStyle: style.microbitImg
+  },
+  circuitPlayground: {
+    id: 'circuit-playground-description',
+    title: applabI18n.makerSetupCircuitPlaygroundTitle(),
+    href: 'https://learn.adafruit.com/introducing-circuit-playground/overview',
+    imgSrc: '../assets/maker/circuit-playground-200.jpg',
+    description: applabI18n.makerSetupCircuitPlaygroundDescription(),
```

Comment:
because this is a `const` here, the calls to `applabI18n.makerSetupCircuitPlaygroundDescription()` are all going to be called on the script load. This might be a problem, actually. I do not know when the locale information is set. It also makes it hard to test since we cannot stub these functions until we have imported this file... and that means we cannot stub these functions because they are actually used. A tricky one. I'd recommend making this a function to return this information and just living with that reality.

Reply:
const turn into function